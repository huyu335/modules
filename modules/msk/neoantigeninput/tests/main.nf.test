// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test neoantigeninput
nextflow_process {

    name "Test Process NEOANTIGENINPUT"
    script "../main.nf"
    process "NEOANTIGENINPUT"

    tag "modules"
    tag "modules_msk"
    tag "neoantigeninput"

    test("neoantigen") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).
        when {
            params {
                folderPath = 'tst'
            }

            process {
                """

                
                
                file(params.folderPath).mkdirs()
                originalFile = file(params.test_data_mskcc['neoantigen']['test4_mutass_0_json'], checkIfExists: true)
                copy_1 = originalFile.copyTo(file("0.json"))
                copy_2 = originalFile.copyTo(file("1.json"))
                copy_1.moveTo(file(params.folderPath + "/0.json"))
                copy_2.moveTo(file(params.folderPath + "/1.json"))

                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_maf'], checkIfExists: true)
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['test4_summ_json'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['test4_muts_json'], checkIfExists: true),
                    file(params.folderPath, checkIfExists: true)
                    ]

                input[2] = [file(params.test_data_mskcc['neoantigen']['winners_hla_txt'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
