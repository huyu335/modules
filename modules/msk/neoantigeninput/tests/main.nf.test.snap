{
    "neoantigen": {
        "content": [
            {
                "0": [
                    [
                    {
                    "id":"test",
                    "single_end":false
                    },
                    "test_patient_test_.json:md5,fede76877e226402bf93ebeabcfb3660"]
                ],
                "1": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ],
                "json": [
                    [{
                    "id":"test",
                    "single_end":false
                    },
                    "test_patient_test_.json:md5,fede76877e226402bf93ebeabcfb3660"]
                ],
                "versions": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-04T11:33:58.514916"
    }
}